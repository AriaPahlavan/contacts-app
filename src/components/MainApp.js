import React, {Component} from 'react';

import CardList from '../components/CardList';
import SreachBox from '../components/SreachBox';
import Scroll from '../components/Scroll';

class MainApp extends Component {
  componentDidMount() {
    this.props.fetchContacts();
  }

  filterKitties = () => {
    const {searchField: query, users: kitties} = this.props;
    return kitties.filter(x => x.name
                                .toLowerCase()
                                .includes(query.toLowerCase()));
    };

  render() {
    const { onSearch, isPending } = this.props;

    return isPending
    ? <h1 className='tc mt7 avenir f1 light-green'>Loading...</h1>
    : (
        <div className='tc'>
          <h1 className="avenir f1 light-green">Your Kitties</h1>
          <SreachBox onSearch={onSearch}/>
          <Scroll>
            <CardList contacts={this.filterKitties()}/>
          </Scroll>
        </div>
      );
  }
}


export default MainApp;
