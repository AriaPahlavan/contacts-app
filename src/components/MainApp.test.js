import {shallow} from 'enzyme';
import React from 'react';
import MainApp from './MainApp';

let wrapper;
let mockProps;

beforeEach(() => {
   mockProps = {
    fetchContacts: jest.fn(),
    searchField: '',
    onSearch: jest.fn(),
    users: [],
    isPending: false
  };

  wrapper = shallow(<MainApp {...mockProps}/>);
});

it('ensures MainApp renders', () => {
  expect.assertions(1);
  expect(wrapper).toMatchSnapshot();
});

it('ensures "no kitties" get filtered correctly', () => {
  expect.assertions(1);
  expect(wrapper.instance().filterKitties()).toEqual([]);
});

it('ensures kitties get filtered correctly', () => {
  expect.assertions(1);

  const randomUser = {
    id: 1,
    name: 'Aria',
    email: 'aria@gmail.com'
  };
  mockProps.users = [randomUser]
  const wrapper2 = shallow(<MainApp {...mockProps}/>);
  expect(wrapper2.instance().filterKitties()).toEqual([randomUser ]);
});
