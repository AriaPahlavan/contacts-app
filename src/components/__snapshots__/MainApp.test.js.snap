// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainApp functionality test ensures MainApp renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainApp
    fetchContacts={[Function]}
    isPending={false}
    onSearch={[Function]}
    searchField=""
    users={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="avenir f1 light-green"
>
          Your Kitties
</h1>,
        <SreachBox
          onSearch={[Function]}
/>,
        <Scroll>
          <CardList
                    contacts={Array []}
          />
</Scroll>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Your Kitties",
          "className": "avenir f1 light-green",
        },
        "ref": null,
        "rendered": "Your Kitties",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onSearch": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <CardList
            contacts={Array []}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "contacts": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="avenir f1 light-green"
>
            Your Kitties
</h1>,
          <SreachBox
            onSearch={[Function]}
/>,
          <Scroll>
            <CardList
                        contacts={Array []}
            />
</Scroll>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your Kitties",
            "className": "avenir f1 light-green",
          },
          "ref": null,
          "rendered": "Your Kitties",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onSearch": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <CardList
              contacts={Array []}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "contacts": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
